pandas/core/tools/datetimes.py

```diff
@@ -220,6 +222,10 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
         raise TypeError('arg must be a string, datetime, list, tuple, '
                         '1-d array, or Series')
 
+    # warn if passing timedelta64, raise for PeriodDtype
+    # NB: this must come after unit transformation
+    arg = dtype_conversions(arg, False, has_format=format is not None)[0]
```

Comment:
all this ``maybe_convert_dtype`` can you pass the 2nd arg as a kwarg

Reply:
done